src/app/ReadHandler.cpp

```diff
@@ -61,6 +62,54 @@ ReadHandler::ReadHandler(ManagementCallback & apCallback, Messaging::ExchangeCon
     mSessionHandle.Grab(mExchangeCtx->GetSessionHandle());
 }
 
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+ReadHandler::ReadHandler(ManagementCallback & apCallback) :
+    mExchangeCtx(*this), mManagementCallback(apCallback), mOnConnectedCallback(HandleDeviceConnected, this),
+    mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+{
+    mInteractionType = InteractionType::Subscribe;
+    mFlags.ClearAll();
+}
+
+void ReadHandler::ResumeSubscription(CASESessionManager & caseSessionManager,
+                                     SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo)
+{
+    mSubscriptionId          = subscriptionInfo.mSubscriptionId;
+    mMinIntervalFloorSeconds = subscriptionInfo.mMinInterval;
+    mMaxInterval             = subscriptionInfo.mMaxInterval;
+    SetStateFlag(ReadHandlerFlags::FabricFiltered, subscriptionInfo.mFabricFiltered);
+
+    // Move dynamically allocated attributes and events from the SubscriptionInfo struct into the object pool managed by the IM
+    // engine
+    CHIP_ERROR err;
+    for (size_t i = 0; i < subscriptionInfo.mAttributePaths.AllocatedCount(); i++)
+    {
+        AttributePathParams attributePathParams = subscriptionInfo.mAttributePaths[i].GetParams();
+        err = InteractionModelEngine::GetInstance()->PushFrontAttributePathList(mpAttributePathList, attributePathParams);
+        if (err != CHIP_NO_ERROR)
+        {
+            Close();
+            return;
+        }
+    }
+    for (size_t i = 0; i < subscriptionInfo.mEventPaths.AllocatedCount(); i++)
+    {
+        EventPathParams eventPathParams = subscriptionInfo.mEventPaths[i].GetParams();
+        err = InteractionModelEngine::GetInstance()->PushFrontEventPathParamsList(mpEventPathList, eventPathParams);
+        if (err != CHIP_NO_ERROR)
+        {
+            Close();
+            return;
+        }
+    }
+
+    // Ask IM engine to start CASE session with subscriber
+    ScopedNodeId peerNode = ScopedNodeId(subscriptionInfo.mNodeId, subscriptionInfo.mFabricIndex);
+    caseSessionManager.FindOrEstablishSession(peerNode, &mOnConnectedCallback, &mOnConnectionFailureCallback);
```

Comment:
This will only work for subscribing controllers that mDNS advertise and operate a CASE server.  For a long time (and this may still be true), chip-tool did not do this.  And so many implementations in the field will also not do so.

Reply:
Discussed offline.

This doesn't need to block merge of this PR.  But we should try to communicate amongst ourselves that this feature is on the horizon and will be ineffective for controllers that don't advertise and operate a CASE server.

Reply:
Absolutely. I'll make sure we have some discussions about this in the coming days.